lib/SetAPI/sampleset/SampleSetInterface.py

```diff
@@ -93,32 +79,30 @@ def create_sample_set(self, ctx, params):
                     "KBaseAssembly.SingleEndLibrary",
                     "KBaseFile.SingleEndLibrary",
                 ]
+            sample_refs = []
             for reads_ref in sample_ids:
+                # TODO: batch the ws calls instead of doing one sample at a time
                 reads_info = self.ws_client.get_object_info3(
                     {"objects": [{"ref": reads_ref}]}
-                )
-                obj_type = reads_info["infos"][0][2].split("-")[0]
+                )["infos"][0]
+                sample_refs.append(info_to_ref(reads_info))
+                obj_type = reads_info[2].split("-")[0]
                 if obj_type not in lib_type:
                     raise ValueError(
-                        "Library_type mentioned : {0}. Please add only {1} typed objects in Reads fields".format(
-                            params["Library_type"], params["Library_type"]
-                        )
+                        f"Library_type mentioned: {params['Library_type']}. Please add only {params['Library_type']} typed objects in Reads fields"
                     )
 
             ## Code to Update the Provenance; make it a function later
-            provenance = [{}]
-            if "provenance" in ctx:
-                provenance = ctx["provenance"]
+            provenance = ctx.get("provenance", [{}])
             # add additional info to provenance here, in this case the input data object reference
-            provenance[0]["input_ws_objects"] = [
-                self._ws_get_ref(params["ws_id"], sample) for sample in sample_ids
-            ]
```

Comment:
We already have this information from the earlier info call, so just collect it then instead of faffing around with another call here.